### PR TITLE
Use RegexAst::Literal instead of const field on strings for fast enums

### PR DESCRIPTION
**Pull request by hudson-ai**

### CODE REVIEW

**Comment by hudson-ai** [comment]
@mmoskal kind of messed up the base of this pr as I had to merge main back into yours

File: parser/src/json/schema.rs [hunk 1]
```diff
@@ -126,6 +128,43 @@ impl Schema {
...
+                maximum: Some(y),
+                ..
+            } if x == y => return Some(RegexNode::Literal(x.to_string())),
+            Schema::String {
+                // TODO: check that min/max don't break the literal and all the other constraints
+                // or just convert this whole node to a regex
+                min_length: 0,
+                max_length: None,
+                regex: Some(RegexAst::Literal(s)),
+            } => &regex_syntax::escape(&serde_json::to_string(s).unwrap()),
```
Comment (hudson-ai) [inline parser/src/json/schema.rs:146]: This is the bit that makes it a bit tricky to do this for general string regexes -- json quoting

Comment (mmoskal) [reply parser/src/json/schema.rs:146]: I'll add AST node to escape. Shouldn't be hard.

**Comment by hudson-ai** [comment]
Closes #84

**Pull request merged**